src/services/matrix.rs

```diff
@@ -1274,13 +1274,13 @@ pub mod matrix {
     }
 
     pub async fn login(
-        homeserver: String,
-        username: String,
-        password: String,
+        homeserver: &String,
```

Comment:
It's usually better practice to accept a `&str` instead of `&String` as `String` can automatically deref to `&str` but if you want to pass a `&str` it would require to first convert to `String`(allocating memory in the process). Tip: Usually seeing a `String` in a parameter should be a bit of a red flag.